date_range/wizard/date_range_generator.py

```diff
@@ -26,37 +35,162 @@ def _default_company(self):
+    def _generate_intervals(self):
+        """Generate a list of dates representing the intervals.
+
+        The last date only serves to compute the end date of the last interval.
+        """
+        if not self.date_end and not self.count:
+            raise ValidationError(
+                _("Please enter an end date, or the number of ranges to "
+                  "generate."))
+        kwargs = dict(
+            freq=int(self.unit_of_time),
+            interval=self.duration_count,
+            dtstart=self.date_start
+        )
+        if self.date_end:
+            kwargs["until"] = self.date_end
+        else:
+            kwargs["count"] = self.count
+        vals = list(rrule(**kwargs))
+        if not vals:
+            raise UserError("No ranges to generate with these settings")
+        # Generate another interval to fetch the last end date from
+        vals.append(
+            list(
+                rrule(
+                    freq=int(self.unit_of_time),
+                    interval=self.duration_count,
+                    dtstart=vals[-1].date(),
+                    count=2,
+                ))[-1])
+        return vals
+
+    @api.multi
+    def generate_names(self, vals):
+        """Generate the names for the given intervals"""
         self.ensure_one()
-        vals = rrule(freq=self.unit_of_time, interval=self.duration_count,
-                     dtstart=self.date_start,
-                     count=self.count+1)
-        vals = list(vals)
-        date_ranges = []
-        count_digits = len(str(self.count))
+        return self._generate_names(vals, self.name_expr, self.name_prefix)
+
+    @staticmethod
+    def _generate_names(vals, name_expr, name_prefix):
+        """Generate the names for the given intervals and naming parameters"""
+        names = []
+        count_digits = len(str(len(vals)))
         for idx, dt_start in enumerate(vals[:-1]):
             date_start = dt_start.date()
             # always remove 1 day for the date_end since range limits are
             # inclusive
             date_end = vals[idx+1].date() - relativedelta(days=1)
+            index = "%0*d" % (count_digits, idx + 1)
+            if name_expr:
+                try:
+                    names.append(safe_eval(
+                        name_expr, {
+                            "date_end": date_end,
+                            "date_start": date_start,
+                            "index": index,
+                        }))
+                except SyntaxError as e:
+                    raise ValidationError(_("Invalid name expression: %s") % e)
+            elif name_prefix:
+                names.append(name_prefix + index)
+            else:
+                raise ValidationError(
+                    _("Please set a prefix or an expression to generate "
+                      "the range names."))
+        return names
+
+    def test_name_expr(self):
+        """Button on the form to preview the generated names"""
```

Comment:
same comment as https://github.com/OCA/server-ux/pull/347/files#r635903610

Reply:
In this case, the button returns all names to generate, with the accurate zero fill on the index. I think that unlike the case on the date range type, we lose value with your suggestion here.

Reply:
Oh maybe I can just run that code in the compute method.